lib/spack/spack/spec.py

```diff
@@ -3770,9 +3770,6 @@ def eq_node(self, other):
     def _cmp_iter(self):
         """Lazily yield components of self for comparison."""
 
-        for item in self._cmp_node():
```

Comment:
the root should go first, otherwise the order on concrete specs is which is by dag hash, which is equivalent to randomization.